generic_tracers/generic_COBALT.F90

```diff
@@ -2702,58 +2705,68 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
     ! captures mixing on time scale of 1 to a few days.
     ! de Boyer-Montegut reference:  https://doi.org/10.1029/2004JC002378
     !
-    do j = jsc, jec ; do i = isc, iec   !{
+    pha_all_same = all(photo_acc_dpth == photo_acc_dpth(isc,jsc))
```

Comment:
@theresa-morrison, I now understand why the regression testing crashed. Since photo_acc_dpth isn't being passed at this stage (as the MOM6 side hasn't been updated), the array is never declared. This leads to an out-of-bounds error when performing the all_same check outside the if (present(photo_acc_dpth)) loop. To resolve this, you can move the all_same check inside the if loop.

Reply:
done!